### ISSUE TITLE
Add `default` callback to `Encoder`/`encode`

### ISSUE THREAD

**Issue by jcrist** [jcrist/msgspec#20]
To support serializing arbitrary objects, it'd be useful to add a `default` callback (mirroring the option in `json.dumps`, `orjson.dumps`, ...). If provided, this would be called on unsupported objects, and would be expected to return a supported object.

```python
def default(obj: Any) -> Any:  # where the out Any is a supported type
    ...
```

This would allow for converting unsupported types into supported types lazily, as well as fancier things like serializing unsupported large objects out-of-band.

**Comment by jcrist** [comment]
cc @jakirkham, @madsbk, @quasiben, @mrocklin.

**Comment by jcrist** [comment]
See #21.

**Comment by jcrist** [comment]
This has now been released as part of 0.2.0, and is available on PyPI. Please let me know how it works out for you, or if you have any further requests/questions.
